### DISCUSSION TITLE
Roger beep tone frequency

### DISCUSSION THREAD

**Discussion by AndrewTXJ** [egzumer/uv-k5-firmware-custom#355] [Q&A] [answered]
How to change the roger beep tone frequency, i prefer the original tone, this firmware seems to change it to higher pitch, but when i am trying to find the correct file to modify, no luck, i am almost zero knowledge on coding🤣🤣🤣.

**Comment by gartnl** [comment]
You can find it in BK4819.c , function BK4819_PlayRogerNormal()
It defaults to a high pitched two-tone motorola style beep which makes you sound like a PMR user ...
Should be easy to change to a single 100ms beep.

    static void BK4819_PlayRogerNormal(void)
    {
  
	  const uint32_t tone1_Hz = 700;
  
  
	  BK4819_EnterTxMute();
	  BK4819_SetAF(BK4819_AF_MUTE);
  
	  BK4819_WriteRegister(BK4819_REG_70, BK4819_REG_70_ENABLE_TONE1 | (66u << BK4819_REG_70_SHIFT_TONE1_TUNING_GAIN));
  
	  BK4819_EnableTXLink();
	  SYSTEM_DelayMs(50);
  
	  BK4819_WriteRegister(BK4819_REG_71, scale_freq(tone1_Hz));
  
	  BK4819_ExitTxMute();
	  SYSTEM_DelayMs(100);
	  BK4819_EnterTxMute();
  
	  BK4819_WriteRegister(BK4819_REG_70, 0x0000);
	  BK4819_WriteRegister(BK4819_REG_30, 0xC1FE);   // 1 1 0000 0 1 1111 1 1 1 0
    }

**Reply by AndrewTXJ** [reply]
what is the original quansheng roger tone? i heard that it is more lower tone when i started to use my K6 in stock firmware.

**Reply by gartnl** [reply]
I really have no idea. I like the 700 Hz tone so I chose that.

**Comment by ErikS-web** [comment]
It is not adjustable in the menu.

Tip: If it is of importance to you. 
You can use an [UVMOD-page](https://whosmatt.github.io/uvmod/) and configure the Roger-tones and use that firmware.

![Knipsel](https://github.com/egzumer/uv-k5-firmware-custom/assets/148579604/cd307220-5930-406b-8def-e37ff464f473)

**Reply by egzumer** [reply]
Don't recommend UVMOD. It is outdated, currently useless tool. People later come and ask why the mod they apply doesn't work, or what to do if it shows memory overflow.

**Reply by gartnl** [reply]
I've noticed on a HAM forum people get confused choosing between all the different firmware mods. And are comparing mods and options, turning it in to some kind of competition. They e.g. do not understand that UVMOD only works when modding some settings in the original QS firmware. And that every mod has it's own focus and own settings. 
The fact that many users don't have a HAM background does not help either. If it was up to the HAM's a rogertone should be a single short beep. The fake 'frog' (DTMF) is not done. But is part of the original firmware and many users seem to like it.

It is easy enough to make settings like this user selectable ( and reading the original code it ones was ) but, and that is a BIG BUT:
once you start doing that another user comes along and wants to change the length of the beeps, another one desires three beeps, I desire one beep and no frog  and so on. So where does it end?

Sometimes you just have to be content with what is offered. Choose whatever mod suits you best and stick with it. Or change it yourself.

**Comment by AndrewTXJ** [comment, marked as answer]
Thanks everyone, i found it

`#else
		// motorola type
		const uint32_t tone1_Hz = 440;
		const uint32_t tone2_Hz = 587;
	#endif`

i did not bother to find out what the original tone was, i basically change it to a tone i like, 440 and 587, which is A4 and D5 in music, more lower, for me, it abit pleasure to listen (although i can't hear it execpt the one who i call will hear this🤣🤣🤣), higher tone sometime will getting more attention but lower tone seems to not (just my own thought, don't know whether it is true🤣🤣🤣).

Thanks again, everyone.🙇🏻‍♀️🙇🏻‍♀️🙇🏻‍♀️

**Reply by Ehopz** [reply]
i have no idea how to code, do you have instructions on how to do this. it would be greatly appreciated.
